samples/namespacescope-sample/createUIDefinition.json

```diff
@@ -0,0 +1,247 @@
+								"showStepMarkers": false,
+								"toolTip": "Specify VM count",
+								"constraints": {
+									"required": false
+								},
+								"visible": true
+							}
+						],
+						"visible": "[basics('createNewCluster')]"
+					}
+				]
+			},
+			{
+				"name": "extensionDetails",
+				"label": "Cluster Extension Details",
+				"elements": [
+					{
+						"name": "extensionResourceName",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Cluster extension resource name",
+						"defaultValue": "",
+						"toolTip": "Only lowercase alphanumeric characters are allowed, and the value must be 6-30 characters long.",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9]{6,30}$",
+							"validationMessage": "Only lowercase alphanumeric characters are allowed, and the value must be 6-30 characters long."
+						},
+						"visible": true
+					},
+					{
+						"name": "targetNamespace",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Installation namespace",
+						"defaultValue": "default",
+						"toolTip": "The Kubernetes namespace in which the extension will be installed. If the namespace does not exist, it will be created. Extension resource name is used as default value",
+						"constraints": {
+						  "required": true,
+						  "regex": "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$",
+						  "validationMessage": "Only lowercase alphanumeric characters or '-' ara allowed, and must start and end with an alphanumeric character."
+						}
+					  },
+					{
+						"name": "title",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Azure Vote Extension Title",
+						"toolTip": "This will replace the title's value inside your values.yaml",
+						"defaultValue": "Azure Vote App",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9A-Z ]{1,30}$",
+							"validationMessage": "Only alphanumeric characters are allowed, and the value must be 6-30 characters long."						 
+						}
+					},
+					{
+						"name": "value1",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Azure Vote Extension Option 1",
+						"toolTip": "This will replace the value1's value inside your values.yaml",
+						"defaultValue": "Cats",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9A-Z]{1,30}$",
+							"validationMessage": "Only alphanumeric characters are allowed, and the value must be 6-30 characters long."						 
+						}
+					},
+					{
+						"name": "value2",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Azure Vote Extension Option 2",
+						"toolTip": "This will replace the value2's value inside your values.yaml",
+						"defaultValue": "Dogs",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9A-Z]{1,30}$",
+							"validationMessage": "Only alphanumeric characters are allowed, and the value must be 6-30 characters long."						 
+						}
+					},
+					{
+						"name": "autoUpgradeMinorVersion",
```

Comment:
>autoUpgradeMinorVersion [](http://example.com/codeflow?start=15&length=23)

Should we move all platform specific parameters to top, followed by application parameters?